### PR TITLE
DHFPROD-5707: added focus style to MLButton component

### PR DESCRIPTION
**Pull request by jyangML**
### Description
Description in title. (Title edited 12/3/2020)
Also see first comment.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by jyangML** [comment]
For reviewers: I opted to set the background and outline color in :focus the same as the hover color (which is #5B69AF), but this also made any outlines very faint.  However, when you do tab over, it's immediately noticeable that the button is in focus, so I've decided that the AC is satisfied for now.  Just be sure to take a look before approving this PR.

**Comment by jyangML** [comment]
Just talked to Kevin, and he's checked off on this :focus style.

**Comment by jyangML** [comment]
Note: the changes in this PR has been updated to include all MLButton components.

**Review by wooldridge — CHANGES_REQUESTED** [review]
Nice work, I'm seeing the standalone buttons taking focus with outline styling across the app. 

I think there should be some kind of testing for button focus in the app, similar to what you have for tools in toolbar.test.tsx. How about we add a representative test for buttons in the Run tile in flows.test.tsx, and verify that the Create Flow button takes focus. With that button focused, also test that pressing Return displays the expected New Flow dialog.

A couple fixes for toolbar.test.tsx, change the two instances of Typescript "int" to "number". Also distinguish the two tests at the end in the descriptions, call the second one: "verify arrow key selection of tile icons"

**Comment by jyangML** [comment]
new changes:
fixed names in toolbar.test.tsx
added a test in flows.test.tsx
cleaned up code for Create Flow button in flows.tsx

**Comment by jyangML** [comment]
fixed an issue where you can't tab out of the Create Flow button

**Review by wooldridge — APPROVED** [review]
Looks great, here is some manual testing of the keyboard nav:  https://drive.google.com/file/d/1o_2bq7Gl5XE0mLqicvojnOzhSlSbp4Gh/view

Unit tests pass.

**Comment by wooldridge** [comment]
Reviewers, note that contrary to the N/Aed checkboxes above, there were tests added.

**Comment by Sanjeevani19** [comment]
> Reviewers, note that contrary to the N/Aed checkboxes above, there were tests added.

**Comment by Sanjeevani19** [comment]
> Reviewers, note that contrary to the N/Aed checkboxes above, there were tests added.

@wooldridge @jyangML Just for consistency purpose since unit tests are there I have enable added tests checkbox.

**Review by Sanjeevani19 — APPROVED** [review]

File: marklogic-data-hub-central/ui/src/components/flows/flows.test.tsx [hunk 1]
```diff
@@ -185,4 +185,32 @@ describe('Flows component', () => {
...
+        // button should be focusable
+        // verified by tabbing away and tabbing back
+        userEvent.tab();
+        expect(flowButton).not.toHaveFocus();
+        userEvent.tab({shift: true});
+        expect(flowButton).toHaveFocus();
+
+        // pressing enter on button should bring up New Flow dialogue box
+        fireEvent.keyDown(flowButton, { key: 'Enter', code: 'Enter' });
+        expect(getByText("New Flow")).toBeInTheDocument();
```
Comment (briantang) [inline marklogic-data-hub-central/ui/src/components/flows/flows.test.tsx:212]: @jyangML Seeing the unit test failing on this line. Can you take a look?

**Review by briantang — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/flows/flows.test.tsx (hunk 1 above)
Comment (briantang) [reply marklogic-data-hub-central/ui/src/components/flows/flows.test.tsx:212]: Disregard this, did not have the latest changes. Looks good now.

**Review by briantang — APPROVED** [review]
Looks good, tests pass.

**Pull request merged**